provider/provider.go

```diff
@@ -236,7 +241,12 @@ func (p *Provider) FetchResources(ctx context.Context, request *cqproto.FetchRes
 			return nil
 		})
 	}
-	return g.Wait()
+	if err := g.Wait(); err != nil {
+		// can we wrap multiple errors here or should we just join them?
+		return err
+	}
```

Comment:
no you cant you can use go-multierror and append to it but you would have to do a lock on it. As errgroup returns on first error.